mypy/nodes.py

```diff
@@ -2208,6 +2208,23 @@ def deserialize(cls, data: JsonDict) -> 'TypeInfo':
         return ti
 
 
+class FakeInfo(TypeInfo):
+    # types.py defines a single instance of this class, called types.NOT_READY.
+    # This instance is used as a temporary placeholder in the process of de-serialization
+    # of 'Instance' types. The de-serialization happens in two steps: In the first step,
+    # Instance.type is set to NOT_READY. In the second step (in fixup.py) it is replaced by
+    # an actual TypeInfo. If you see the assertion error below, then most probably something
+    # went wrong during the second step and an 'Instance' that raised this error was not fixed.
+    # Note:
+    # 'None' is not used as a dummy value for two reasons:
+    # 1. This will require around 80-100 asserts to make 'mypy --strict-optional mypy'
+    #    pass cleanly.
+    # 2. If NOT_READY value is accidentally used somewhere, it will be obvious where the value
+    #    is from, whereas a 'None' value could come from anywhere.
+    def __getattr__(self, attr: str) -> None:
```

Comment:
@ilevkivskyi @gvanrossum This should probably be `__getattribute__` instead. Now attributes defined in the body of `TypeInfo` will not trigger an `AssertionError`, as far as I know, and thus `FakeInfo` can mask some actual errors.